lib/puppet/util/network_device/transport/ssh.rb

```diff
@@ -68,9 +68,12 @@ def connect(&block)
   end
 
   def close
-    @channel.close if @channel
-    @channel = nil
-    @ssh.close if @ssh
+    begin
+      @channel.close if @channel
+      @channel = nil
+      @ssh.close if @ssh
+    rescue IOError
```

Comment:
I don't think it's worth being a warning, but it might be useful to have the exception logged at info or debug level